### ISSUE TITLE
Issues with Python 2.7.9

### ISSUE THREAD

**Issue by rohitpaulk** [nickstenning/honcho#122]
14 tests fail on Python 2.7.9, all are integration tests. 

```
======================================================================
FAIL: test_start_with_arg (honcho.test.integration.test_start.TestStart)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/paul/experiments/gratipay/honcho/honcho/test/integration/test_start.py", line 24, in test_start_with_arg
    self.assertEqual(ret, 0)
AssertionError: 1 != 0

======================================================================
FAIL: test_start_with_arg_returncode (honcho.test.integration.test_start.TestStart)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/paul/experiments/gratipay/honcho/honcho/test/integration/test_start.py", line 59, in test_start_with_arg_returncode
    self.assertEqual(ret, 42)
AssertionError: 1 != 42

======================================================================
FAIL: test_start_with_multiple_args (honcho.test.integration.test_start.TestStart)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/paul/experiments/gratipay/honcho/honcho/test/integration/test_start.py", line 35, in test_start_with_multiple_args
    self.assertEqual(ret, 0)
AssertionError: 1 != 0

======================================================================
FAIL: test_unicode_decode_error (honcho.test.integration.test_unicode_decode.TestUnicodeDecode)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/paul/experiments/gratipay/honcho/honcho/test/integration/test_unicode_decode.py", line 10, in test_unicode_decode_error
    self.assertEqual(ret, 0)
AssertionError: 1 != 0

----------------------------------------------------------------------
Ran 86 tests in 6.461s

FAILED (failures=14)
```

Honcho doesn't seem to be picking up the environment variables - we're having issues over at https://github.com/gratipay/gratipay.com/issues/3025#issuecomment-67515281

**Comment by fnavarrog** [comment]
Just for completeness, I'll add a different situation where it will dramatically affect working code. In this case, if the ```-f``` parameter comes before the ```start``` command, honcho will endlessly fork new processes.

In a specific case, those working with foreman+honcho in Heroku, will see their startup sequence stop working. Repro files in here, https://gist.github.com/anonymous/6705f438f83e4347850c

Output will look like this
```
17:12:58 web.1  | started with pid 217
17:12:58 web.1  | 17:12:58 web.1  | started with pid 220
17:12:58 web.1  | 17:12:58 web.1  | 17:12:58 web.1  | started with pid 223
17:12:58 web.1  | 17:12:58 web.1  | 17:12:58 web.1  | 17:12:58 web.1  | started with pid 226
17:12:58 web.1  | 17:12:58 web.1  | 17:12:58 web.1  | 17:12:58 web.1  | 17:12:58 web.1  | started with pid 229
17:12:58 web.1  | 17:12:58 web.1  | 17:12:58 web.1  | 17:12:58 web.1  | 17:12:58 web.1  | 17:12:58 web.1  | started with pid 232
```

**Comment by nickstenning** [comment]
Closed by 1319da7 and released as v0.6.4.
